include/pybind11/numpy.h

```diff
@@ -220,6 +219,16 @@ inline bool check_flags(const void* ptr, int flag) {
     return (flag == (array_proxy(ptr)->flags & flag));
 }
 
+template <typename T> struct is_std_array : std::false_type { };
+template <typename T, size_t N> struct is_std_array<std::array<T, N>> : std::true_type { };
+template <typename T> struct is_complex : std::false_type { };
+template <typename T> struct is_complex<std::complex<T>> : std::true_type { };
+
+template <typename T> using is_pod_struct = all_of<
+    std::is_pod<T>, // offsetof only works correctly for POD types
```

Comment:
Nitpicking: `offsetof` only needs `is_standard_layout`, it's numpy that needs the rest of `is_pod`.